azure/scope/identity_test.go

```diff
@@ -377,11 +436,15 @@ func TestGetTokenCredential(t *testing.T) {
 				initObjects = append(initObjects, tt.secret)
 			}
 			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initObjects...).Build()
-			provider, err := NewAzureCredentialsProvider(context.Background(), fakeClient, tt.cluster.Spec.IdentityRef, "")
+
+			mockCtrl := gomock.NewController(t)
+			cache := mock_azure.NewMockCredentialCache(mockCtrl)
+			tt.cacheExpect(cache)
+
+			provider, err := NewAzureCredentialsProvider(context.Background(), cache, fakeClient, tt.cluster.Spec.IdentityRef, "")
 			g.Expect(err).NotTo(HaveOccurred())
-			cred, err := provider.GetTokenCredential(context.Background(), "", tt.ActiveDirectoryAuthorityHost, "")
+			_, err = provider.GetTokenCredential(context.Background(), "", tt.ActiveDirectoryAuthorityHost, "")
```

Comment:
why aren't we asserting the first response object?

Reply:
That value is returned by a mock defined by the test, so it doesn't really mean anything.

Reply:
The main validation of that is covered by verifying the arguments we pass to the `p.cache.GetOrStore...` calls in the `cacheExpect` of each test case.

Reply:
I see that `GetTokenCredential` returns non-nil every time `err` is nil, maybe I'm being extra-defensive and protecting against the possibility that `GetTokenCredential` changes and an undesirable outcome is identified only in the `TokenCredential` object.

Non-blocking.